Chessington.GameEngine/Pieces/Pawn.cs

```diff
@@ -5,12 +5,46 @@ namespace Chessington.GameEngine.Pieces
 {
     public class Pawn : Piece
     {
-        public Pawn(Player player) 
-            : base(player) { }
+        public Pawn(Player player)
+            : base(player)
+        {
+        }
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            var availablePositions = new List<Square>();
+            var currentSquare = board.FindPiece(this);
+            var rowDir = Player == Player.White ? new[] { -1, -2 } : new[] { 1, 2 };
+            
+            var inFrontSquare = Square.At(currentSquare.Row + rowDir[0], currentSquare.Col);
+            if (inFrontSquare.IsSquareValid(board))
+            {
+                availablePositions.Add(inFrontSquare);
+            }
+            else
+            {
+                return Enumerable.Empty<Square>();
+            }
+
+            var inFrontTwoSquares = Square.At(currentSquare.Row + rowDir[1], currentSquare.Col);
+            if (Moved == false && inFrontTwoSquares.IsSquareValid(board))
```

Comment:
I think it's more readable to write `!Moved`, since I can read it more coherently as `not Moved`